src/PackageHelper.js

```diff
@@ -0,0 +1,207 @@
+    get cordova () {
+        // This is a poor way to deeply clone the data, but it's fast and since
+        // it's package.json we can guarantee that it's safe to stringify and
+        // parse again without losing anything
+        return deepFreeze(JSON.parse(JSON.stringify(this.pkgfile.cordova || {})));
+    }
+
+    /**
+     * Returns an object keyed by the name of Cordova dependencies and their
+     * requested versions.
+     *
+     * @returns {{[name: string] : string}} The Cordova dependencies.
+     */
+    get cordovaDependencies () {
+        return deepFreeze(
+            this.cordovaPlatforms
+                .map(p => p.startsWith('cordova-') ? p : `cordova-${p}`)
+                .concat(this.cordovaPlugins)
+                .reduce((deps, cur) => {
+                    deps[cur] = this.pkgfile.devDependencies[cur] || this.pkgfile.dependencies[cur];
+                    return deps;
+                }, {})
+        );
+    }
+
+    /**
+     * Retrieves the list of Cordova platforms for the project.
+     *
+     * @returns {Array<string>} The list of packages.
+     */
+    get cordovaPlatforms () {
+        return deepFreeze(this.cordova.platforms || []);
+    }
+
+    /**
+     * Retrieves the list of Cordova plugins for the project.
+     *
+     * @returns {Array<string>} The list of plugins.
+     */
+    get cordovaPlugins () {
+        return deepFreeze(Object.keys(this.cordova.plugins || {}));
+    }
+
+    /**
+     * Retrieves the configuration variables for a particular plugin.
+     *
+     * @param {string} pluginName - The plugin for which to retrieve variables.
+     * @returns {{[name: string] : string}} The plugin variables.
+     */
+    getPluginVariables (pluginName) {
+        const plugins = this.cordova.plugins || {};
+
+        return deepFreeze(JSON.parse(JSON.stringify(plugins[pluginName] || {})));
+    }
+
+    /**
+     * Adds the specified plugin to the Cordova section of the package file.
+     *
+     * @param {string} platform - The platform to add to the package.
+     */
+    addPlatform (platform) {
+        if (!this.cordovaPlatforms.includes(platform)) {
+            this.pkgfile.cordova = this.pkgfile.cordova || {};
+            this.pkgfile.cordova.platforms = this.pkgfile.cordova.platforms || [];
+
+            this.pkgfile.cordova.platforms.push(platform);
+        }
+    }
+
+    /**
+     * Adds the specified plugin and configuration variables to the Cordova section of the package file.
+     *
+     * @param {string} plugin - The plugin to add to the package.
+     * @param {object} vars - The plugin configuration variables (optional).
+     */
+    addPlugin (plugin, vars = {}) {
+        if (this.cordovaPlugins.includes(plugin)) {
+            // Already have the plugin installed, but might need to merge vars
+            let existing_vars = this.pkgfile.cordova.plugins[plugin];
+
+            this.pkgfile.cordova.plugins[plugin] = Object.assign({}, existing_vars, vars);
```

Comment:
I would have expected this to replace the variables, not merge them. With the current implementation we cannot remove variables. Merging would still be doable, if a little verbose:
```js
const oldVars = pkgHelper.getPluginVariables('cordova-plugin-foo')
pkgHelper.addPlugin('cordova-plugin-foo', Object.assign({}, oldVars, newVars))
```
I couldn't tell whether we have to merge often. If so, we should probably provide an option for that.